dashboard/app/controllers/admin_search_controller.rb

```diff
@@ -80,18 +80,23 @@ def show_pilot
     @emails = User.where(id: user_ids).pluck(:email)
   end
 
+  # Parses newline separated emails, ignores commas and whitespace
   def add_to_pilot
-    email = params[:email]
+    emails = params[:email]
     pilot_name = params[:pilot_name]
-    return head :bad_request unless Experiment::PILOT_EXPERIMENTS.find {|p| p[:name] == pilot_name}
-    user = User.find_by_email_or_hashed_email(email)
-    if !user
-      flash[:alert] = "An account with the email address #{email} does not exist"
-    elsif user.student?
-      flash[:alert] = "Cannot add a student to the pilot"
-    else
-      SingleUserExperiment.find_or_create_by!(min_user_id: user.id, name: pilot_name)
-      flash[:notice] = "Successfully added #{email} to #{pilot_name}!"
+    email_array = emails.split("\n").map(&:strip)
+    email_array.each do |email|
+      email = email.gsub(/[\s,]/, "")
+      return head :bad_request unless Experiment::PILOT_EXPERIMENTS.find {|p| p[:name] == pilot_name}
```

Comment:
Should this return happen in the each block? Seems like it should happen before

Reply:
Each email address has to be validated on its own, and should not prevent the other addresses from being processed. With that in mind, does it make sense to keep here?

Reply:
Doesn't this line validate the pilot name? That shouldn't have to happen for each email address, right? It should just have to happen once at the beginning

Reply:
(facepalm) yes, thank you. Will do!